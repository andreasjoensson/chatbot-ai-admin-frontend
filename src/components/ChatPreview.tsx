import React from 'react';
import chatbotai from '../images/logo/ChatbotAI-light.png';

type Message = {
  message: string;
  isBot: boolean;
};

type ChatPreviewProps = {
  logo: string | null;
  messages: Message[];
  userMessageColor: string;
  botMessageColor: string;
  userTextColor: string;
  botTextColor: string;
};

const ChatPreview: React.FC<ChatPreviewProps> = ({
  logo,
  messages,
  userMessageColor,
  botMessageColor,
  userTextColor,
  botTextColor,
}: ChatPreviewProps) => {
  return (
    <div id="chat-container" className="bg-white border">
      <div className="flex items-center justify-between rounded-t-lg  border-b  p-4 text-white">
        {logo ? (
          <img
            src={logo}
            width={125}
            height={125}
            alt="Picture of the author"
          />
        ) : (
          <img
            src={chatbotai}
            width={125}
            height={125}
            alt="Picture of the author"
          />
        )}
        <button
          id="close-chat"
          className="text-black hover:text-gray-400 focus:text-gray-400 focus:outline-none"
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-6 w-6"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
              d="M6 18L18 6M6 6l12 12"
            ></path>
          </svg>
        </button>
      </div>
      <div id="chatbox" className="h-80 overflow-y-auto p-4">
        {messages.map((message: Message) =>
          message.isBot ? (
            <div key={message.message} className="mb-2">
              <p
                style={{
                  backgroundColor: userMessageColor,
                  color: userTextColor,
                }}
                className="inline-block rounded-lg bg-gray-200 px-4 py-2 text-black"
              >
                {message.message}
              </p>
            </div>
          ) : (
            <div key={message.message} className="mb-2 text-right">
              <p
                style={{
                  backgroundColor: botMessageColor,
                  color: botTextColor,
                }}
                className="inline-block rounded-lg px-4 py-2 text-white"
              >
                {message.message}
              </p>
            </div>
          ),
        )}
      </div>
      <div className="flex border-t p-4">
        <input
          id="user-input"
          type="text"
          placeholder="Skriv en besked..."
          className="w-full rounded-l-md border px-3 py-2 focus:outline-none focus:ring-2 focus:ring-blue-500"
        />
        <button
          id="send-button"
          className="rounded-r-md px-4 py-2 text-black transition duration-300 hover:bg-blue-600"
        >
          Send
        </button>
      </div>
    </div>
  );
};

export default ChatPreview;
