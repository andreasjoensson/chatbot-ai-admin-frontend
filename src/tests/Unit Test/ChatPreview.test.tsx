import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import ChatPreview from '../../components/ChatPreview';

describe('ChatPreview', () => {
  const defaultProps = {
    logo: 'logo.png',
    messages: [
      { message: 'Hello', isBot: true },
      { message: 'Hi', isBot: false },
    ],
    userMessageColor: '#FFFFFF',
    botMessageColor: '#000000',
    userTextColor: '#FFFFFF',
    botTextColor: '#000000',
  };

  it('should render without errors', () => {
    const { container } = render(<ChatPreview {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the logo when provided', () => {
    const { getByAltText } = render(<ChatPreview {...defaultProps} />);
    const logoImage = getByAltText('Picture of the author');
    expect(logoImage).toBeInTheDocument();
  });

  it('should render the default logo when not provided', () => {
    const { getByAltText } = render(
      <ChatPreview {...defaultProps} logo={null} />,
    );
    const defaultLogoImage = getByAltText('Picture of the author');
    expect(defaultLogoImage).toBeInTheDocument();
  });

  it('should render bot messages with correct styles', () => {
    const { getByText } = render(<ChatPreview {...defaultProps} />);
    const botMessage = getByText('Hello');
    expect(botMessage).toHaveStyle('background-color: #000000');
    expect(botMessage).toHaveStyle('color: #000000');
  });

  it('should render user messages with correct styles', () => {
    const { getByText } = render(<ChatPreview {...defaultProps} />);
    const userMessage = getByText('Hi');
    expect(userMessage).toHaveStyle('background-color: #FFFFFF');
    expect(userMessage).toHaveStyle('color: #FFFFFF');
  });
});
