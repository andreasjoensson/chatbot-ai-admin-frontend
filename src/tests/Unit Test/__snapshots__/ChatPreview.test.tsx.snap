// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatPreview should render without errors 1`] = `
<div>
  <div
    class="bg-white border"
    id="chat-container"
  >
    <div
      class="flex items-center justify-between rounded-t-lg  border-b  p-4 text-white"
    >
      <img
        alt="Picture of the author"
        height="125"
        src="logo.png"
        width="125"
      />
      <button
        class="text-black hover:text-gray-400 focus:text-gray-400 focus:outline-none"
        id="close-chat"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="h-80 overflow-y-auto p-4"
      id="chatbox"
    >
      <div
        class="mb-2"
      >
        <p
          class="inline-block rounded-lg bg-gray-200 px-4 py-2 text-black"
          style="background-color: rgb(0, 0, 0); color: rgb(0, 0, 0);"
        >
          Hello
        </p>
      </div>
      <div
        class="mb-2 text-right"
      >
        <p
          class="inline-block rounded-lg px-4 py-2 text-white"
          style="background-color: rgb(255, 255, 255); color: rgb(255, 255, 255);"
        >
          Hi
        </p>
      </div>
    </div>
    <div
      class="flex border-t p-4"
    >
      <input
        class="w-full rounded-l-md border px-3 py-2 focus:outline-none focus:ring-2 focus:ring-blue-500"
        id="user-input"
        placeholder="Skriv en besked..."
        type="text"
      />
      <button
        class="rounded-r-md px-4 py-2 text-black transition duration-300 hover:bg-blue-600"
        id="send-button"
      >
        Send
      </button>
    </div>
  </div>
</div>
`;
